File: python/examples/thinking/basic.py
from mirascope import llm


@llm.call("google/gemini-3-pro", thinking={"level": "high", "include_thoughts": True})
def solve(problem: str) -> str:
    return problem


response = solve("What is the first prime number that contains 42 as a substring?")
print(response.pretty())

File: typescript/examples/thinking/basic.ts
import { llm } from "mirascope";

const solve = llm.defineCall<{ problem: string }>({
  model: "google/gemini-2.5-flash",
  thinking: { level: "high", includeThoughts: true },
  template: ({ problem }) => problem,
});

const response = await solve({
  problem: "What is the first prime number that contains 42 as a substring?",
});
console.log(response.pretty());
